internal/pkg/errors/template_errors.go

```diff
@@ -167,32 +175,41 @@ func (p *PackTemplateError) enhanceNPE() {
 			atPackName := parts[1]
 			rootPackName := p.tplctx.Name()
 
-			if atPackName == rootPackName {
+			if atPackName == rootPackName || atPackName == "my" {
 				atPackName = ""
 			}
 
 			p.Suggestions = []string{
+				// TODO: This error will need to be modified if parser-v1 becomes unsupported.
 				fmt.Sprintf(
-					"The legacy %q syntax should be updated to use `var %q .%s`.",
+					"The legacy %q syntax should be updated to use `var %q .%s`. You can run legacy packs unmodified by using the `--parser-v1` flag",
 					p.at,
 					parts[2],
 					atPackName,
 				),
 			}
 		}
-
 	}
 }
 
+func (p *PackTemplateError) isV2Error() bool {
+	return strings.HasSuffix(p.Err.Error(), "not implemented for nomad-pack's v1 syntax")
+}
+
+func (p *PackTemplateError) enhanceV2Error() {
+	p.Suggestions = []string{"Verify that the `--parser-v1` flag is not set when running this pack."}
```

Comment:
Should this append to `p.Suggestions` so it doesn't clobber it?

```suggestion
	p.Suggestions = append(p.Suggestions, "Verify that the `--parser-v1` flag is not set when running this pack.")
```

Reply:
It's safe right now, since the error conditions are mutually exclusive; but we could initialize the slice and switch to `append` if that's better. 🤔